crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -542,6 +542,27 @@ impl Backend {
                     *self.fork.write() = Some(fork);
                     *self.env.write() = env;
                 } else {
+                    // If rpc url is unspecified, then update the fork with the new block number and
```

Comment:
this makes sense, can you please externalize this code (which is done also in the case of rpc url specified) in a helper fn / closure and reuse?

https://github.com/foundry-rs/foundry/blob/f399820c661ee6f00f473303d9022f592c2eb00f/crates/anvil/src/eth/backend/mem/mod.rs#L531-L534